pypgstac/pypgstac/load.py

```diff
@@ -1,246 +1,501 @@
+    if (
+        filename is None
+        or filename == "-"
+        or filename == "stdin"
+        or filename == "stdout"
+    ):
+        if "r" in mode:
+            stream = sys.stdin
+        else:
+            stream = sys.stdout
+        if "b" in mode:
+            fh = stream.buffer
+        else:
+            fh = stream
+        close = False
+    else:
+        fh = open(filename, mode, *args, **kwargs)
+        close = True
+
+    try:
+        yield fh
+    finally:
+        if close:
+            try:
+                fh.close()
+            except AttributeError:
+                pass
+
+
+def name_array_asdict(na: List) -> dict:
+    """Create a dict from a list with key from name field."""
+    out = {}
+    for i in na:
+        out[i["name"]] = i
+    return out
+
+
+def name_array_diff(a: List, b: List) -> List:
+    """Diff an array by name attribute."""
+    diff = dict_minus(name_array_asdict(a), name_array_asdict(b))
+    vals = diff.values()
+    return [v for v in vals if v != {}]
+
+
+def dict_minus(a: dict, b: dict) -> dict:
+    """Get a recursive difference between two dicts."""
+    out: dict = {}
+    for key, value in b.items():
+        if isinstance(value, list):
+            try:
+                arraydiff = name_array_diff(a[key], value)
+                if arraydiff is not None and arraydiff != []:
+                    out[key] = arraydiff
+                continue
+            except KeyError:
+                pass
+            except TypeError:
+                pass
+
+        if value is None or value == []:
+            continue
+        if a is None or key not in a:
+            out[key] = value
+            continue
+
+        if a.get(key) != value:
+            if isinstance(value, dict):
+                out[key] = dict_minus(a[key], value)
+                continue
+            out[key] = value
+
+    return out
+
+
+def read_json(file: str) -> Iterable:
+    """Load data from an ndjson or json file."""
+    open_file: Any = open_std(file, "r")
+    with open_file as f:
+        # Try reading line by line as ndjson
+        try:
+            for line in f:
+                line = line.strip().replace("\\\\", "\\").replace("\\\\", "\\")
+                yield orjson.loads(line)
+        except JSONDecodeError:
+            # If reading first line as json fails, try reading entire file
+            logging.info("First line could not be parsed as json, trying full file.")
+            try:
+                f.seek(0)
+                json = orjson.loads(f.read())
+                if isinstance(json, list):
+                    for record in json:
+                        yield record
+                else:
+                    yield json
+            except JSONDecodeError:
+                logging.info("File cannot be read as json")
+                raise
+
+
+@dataclass
+class Loader:
+    """Utilities for loading data."""
+
+    db: PgstacDB
+    minimizeproj: Optional[bool] = False
```

Comment:
Is this used?

Reply:
Ah, I stripped that out. When I was testing everything I could use to compress, I did have code that would look up the EPSG code if something had a proj:wkt_text and swap it out with just the reference to the code. I'll get rid of the flag.